portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -28,15 +33,33 @@
 @WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  private final List<String> testComments = new ArrayList<String>(Arrays.asList( 
-    "This is an example comment.",
-    "Another example comment...",
-    "Wow! Yet another example comment?"));
-
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // load comments from database
+    Query query = new Query("Comment");
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+    // return list of results
+    List<String> comments = new ArrayList<>();
+    for (Entity entity : results.asIterable()) {
+      String content = (String) entity.getProperty("content");
```

Comment:
According to the documentation of getProperty (https://cloud.google.com/appengine/docs/standard/java/javadoc/com/google/appengine/api/datastore/PropertyContainer.html#getproperty): "The value returned may not be the same type as originally set via setProperty(java.lang.String, java.lang.Object)."

Have you checked that? Should we check that it is a String before we do the typecast?

Reply:
Should I throw an exception if the object isn't a String and if so, what kind would be best?

Reply:
Now you are ignoring any other content which is fine for now! Thanks!
In practice I would not throw an exception here, but rather log an error. If we throw an exception, then it is broken for the user. If we log an error, then we would see the issue in our logs and we could fix it, but users still might get "some" content.